fastai/text/data.py

```diff
@@ -13,44 +13,106 @@
 text_extensions = {'.txt'}
 
 class LanguageModelPreLoader(Callback):
-    "Transforms the texts in `dataset` in a stream for language modelling."
-    def __init__(self, dataset:LabelList, lengths:Collection[int]=None, bs:int=64, bptt:int=70, backwards:bool=False, 
-                 shuffle:bool=False, drop_last:bool=False):
-        self.dataset,self.bs,self.bptt,self.backwards = dataset,bs,bptt,backwards
-        self.shuffle,self.drop_last = shuffle,drop_last
-        self.lengths = np.array(ifnone(lengths, [len(o) for o in dataset.x.items]))
-        self.n = self.lengths.sum() // self.bs
-        
-    def __len__(self):
-        if self.item is not None:    return 1
-        if (self.n-1)%self.bptt == 0 or self.drop_last: return ((self.n-1) // self.bptt) * self.bs
-        return ((self.n-1) // self.bptt + 1) * self.bs
+    "Create a dataloader with bptt slightly changing."
     
+    class CircularIndex():
```

Comment:
I'm not seeing why this class is so complicated: foward can be a simple attribute here, why are you using a private variable and a setter?

Reply:
idx = self.idx
using idx is faster than using self.idx even if it is only used twice in the forwards use.
https://stackoverflow.com/questions/12397984/why-is-local-variable-access-faster-than-class-member-access-in-python

%timeit confirms this